app/models/manageiq/providers/amazon/cloud_manager/event_catcher/stream.rb

```diff
@@ -175,10 +175,21 @@ def parse_event(message)
       # CloudWatch with CloudTrail for API requests Events
       event["eventType"]    = event.fetch_path("detail", "eventName")
       event["event_source"] = :cloud_watch_api
-    elsif event["detail-type"] == "EC2 Instance State-change Notification"
+    elsif event["detail-type"]
       # CloudWatch EC2 Events
-      state                 = "_#{event.fetch_path("detail", "state")}" if event.fetch_path("detail", "state")
-      event["eventType"]    = "#{event["detail-type"].tr(" ", "_").tr("-", "_")}#{state}"
+      detail_type = event["detail-type"]
+
+      if detail_type == "EC2 Instance State-change Notification"
+        state              = "_#{event.fetch_path("detail", "state")}" if event.fetch_path("detail", "state")
+        event["eventType"] = "#{detail_type.tr(" ", "_").tr("-", "_")}#{state}"
+      elsif ["EBS Volume Notification", "EBS Snapshot Notification"].include?(detail_type)
+        event["eventType"] = event.fetch_path("detail", "event")
+      else
+        # Not recognized event, ignoring...
+        $log.debug("#{log_header} Parsed event from SNS Message not recognized #{event}")
+        return
+      end
+
```

Comment:
@bronaghs Hmm, I suppose I don't know for sure. I was trying to avoid duplicating some code there, but perhaps it's best to play it safe.